### PR TITLE
Add `VenmoClient`,  `VenmoCheckoutRequest` and `VenmoCheckoutResult`

### PR DESCRIPTION
**Pull request by agedd**
### Summary of changes

-  Add `VenmoCheckoutRequest` and `VenmoCheckoutResult` Types [DTMOBILES-779 and  DTMOBILES-781]
-  Add `VenmoClient` class and its relevant properties and methods [DTMOBILES-785] 

Note**: This changes are just part of a shell and a future PR will introduce the implementation 

### Checklist

- [x] Added a changelog entry

- @agedd

### CODE REVIEW

File: Sources/VenmoPayments/VenmoCheckoutRequest.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+public struct VenmoCheckoutRequest {
+    
+    /// The order ID associated with the request
+    public let orderID: String
+    
+    /// Creates an instance of VenmoCheckoutRequest
+    ///  - Parameter orderID: The ID of an order to be approved
+    public init(orderID: String) {
+        self.orderID = orderID
+    }
```
Comment (sshropshire) [inline Sources/VenmoPayments/VenmoCheckoutRequest.swift]:
```suggestion
```
small nitpick: Can we leverage the default constructor here like in `VenmoCheckoutResult`?

File: Sources/VenmoPayments/VenmoClient.swift [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+    
+    // MARK: - Initializers
+    
+    ///  Initialize a VenmoClient to process a Venmo payment
+    ///  - Parameter config: The CoreConfig object
+    init(config: CoreConfig, universalLink: URL) {
+        self.config = config
+        self.universalLink = universalLink
+        self.networkingClient = NetworkingClient(coreConfig: config)
+    }
```
Comment (sshropshire) [inline Sources/VenmoPayments/VenmoClient.swift]:
```suggestion
    ///  Initialize a VenmoClient to process a Venmo payment
    ///  - Parameter config: The CoreConfig object
    convenience init(config: CoreConfig, universalLink: URL) {
        self.init(config: config, universalLink: universalLink, networkingClient: NetworkingClient(coreConfig: config))
    }
```
Take or leave: I noticed that the other constructor is primarily for testing. I'd add this suggestion only if it makes sense to DRY up the initialization logic.

**Review by sshropshire — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,7 +7,9 @@
   * Add deprecated warning message to all public classes and methods
 * VenmoPayments
   * Add `VenmoPayments` module 
-  
+  * Add `VenmoCheckoutRequest` and `VenmoCheckoutResult` type for interacting with `start` method
+  * Add `start(_ request: VenmoCheckoutRequest)` method to `VenmoClient`
```
Comment (jaxdesmarais) [inline CHANGELOG.md]:
```suggestion
  * Add `start(_:)` method to `VenmoClient`
```

File: Sources/VenmoPayments/VenmoClient.swift [hunk 4]
```diff
@@ -1,6 +1,38 @@
 import Foundation
+#if canImport(CorePayments)
```
Comment (jaxdesmarais) [inline Sources/VenmoPayments/VenmoClient.swift:3]:
```suggestion

#if canImport(CorePayments)
```

File: Sources/VenmoPayments/VenmoClient.swift [hunk 5]
```diff
@@ -1,6 +1,38 @@
...
+    // MARK: - Properties
+    
+    private let config: CoreConfig
+    private let universalLink: URL
+    private let networkingClient: NetworkingClient
+    
+    // MARK: - Initializers
+    
+    ///  Initialize a VenmoClient to process a Venmo payment
+    ///  - Parameter config: The CoreConfig object
```
Comment (jaxdesmarais) [inline Sources/VenmoPayments/VenmoClient.swift:18]: Lets add docstrings for the `universalLink` param as well

File: Sources/VenmoPayments/VenmoClient.swift [hunk 6]
```diff
@@ -1,6 +1,38 @@
...
+    /// For internal use for testing/mocking purpose
+    init(config: CoreConfig, universalLink: URL, networkingClient: NetworkingClient) {
+        self.config = config
+        self.universalLink = universalLink
+        self.networkingClient = networkingClient
+    }
+    
+    /// Launch the Venmo checkout flow
+    /// - Parameters:
+    ///     - request: the `VenmoCheckoutRequest` for the transaction
```
Comment (jaxdesmarais) [inline Sources/VenmoPayments/VenmoClient.swift:35]: Lets add docstrings for the returns and throws portion as well

File: Sources/VenmoPayments/VenmoClient.swift [hunk 7]
```diff
@@ -1,6 +1,38 @@
...
+    public func start(_ request: VenmoCheckoutRequest) async throws -> VenmoCheckoutResult {
+        // TODO: - Add logic in a future PR
+        .init(orderID: "test-order-id")
```
Comment (jaxdesmarais) [inline Sources/VenmoPayments/VenmoClient.swift]:
```suggestion
        VenmoCheckoutResult(orderID: "test-order-id")
```

File: UnitTests/VenmoPaymentsTests/VenmoClient_Tests.swift [hunk 8]
```diff
@@ -1,6 +1,20 @@
 import XCTest
+import AuthenticationServices
```
Comment (jaxdesmarais) [inline UnitTests/VenmoPaymentsTests/VenmoClient_Tests.swift]: What's the AuthenticationServices import used for?

Comment (agedd) [reply UnitTests/VenmoPaymentsTests/VenmoClient_Tests.swift]: ahh yea it's not needed, resolved in bf065ba8bcb28a40bb2bd4bdeb6b0d6f84ba26fa

**Review by stechiu — APPROVED** [review]

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**